### PR TITLE
[GOVCMS-6928] Ban outgoing HTTP requests in theme

### PR DESCRIPTION
**Pull request by barbun**
#  Issue
Currently, outgoing HTTP requests can be made utilising either `Drupal::httpClient()` or `GuzzleHttp\Client`. These methods introduce security risks and should be banned.

# Proposed solution
Expand on `phpstan` ruleset (used by `shipshape`) to disallow `GuzzleHttp\Client` namespace and `Drupal::httpClient()` static method.

### CODE REVIEW

**Review by steveworley — CHANGES_REQUESTED** [review]
Two things for this one:

- Rebase against `10.x-develop` to make sure we have the right base
- Create a `phpstan.neon` and update the `shipshape.yml` file to run another sniff with a different configuration, the second configuration will have a lower severity and will be treated as an introduction to get end users time to review the pipeline outputs and update the code as required

Thanks!

**Comment by barbun** [comment]
thanks @steveworley , pushed an update that adds `phpstan-extra.neon` config file to separate the checks with lower severity.

**Comment by simesy** [comment]
What is the security risk out of interest, and how would it differ to using Feeds?

**Comment by barbun** [comment]
> What is the security risk out of interest, and how would it differ to using Feeds?

Thanks @simesy 

The rationale is that a theme should not be making outgoing HTTP requests.

Using Migrate would adhere to Drupal safeguards and also limit data parsing abilities to avoid e.g. malicious executables from being intentionally or unintentionally pulled from a remote at runtime.

**Comment by simesy** [comment]
> The rationale is that a theme should not be making outgoing HTTP requests.

Welcome to GovCMS lets do weird stuff in the theme!

**Comment by simesy** [comment]
I don't have any skin in this game but it seems like a sane/safe thing to do as long as no existing SaaS sites are using this method.

**Comment by barbun** [comment]
> I don't have any skin in this game but it seems like a sane/safe thing to do as long as no existing SaaS sites are using this method.

Good pick up :-) 
Yeah, we are setting severity to low for now, which shouldn't block deployments. I am also doing some testing with the SaaS projects before we merge this.

**Review by steveworley — APPROVED** [review]

File: drupal/settings/all.settings.php [hunk 1]
```diff
@@ -177,3 +177,6 @@ public function __toString() {
 if (getenv('LAGOON')) {
   $settings['trusted_host_patterns'][] = '.*';
 }
+
+// Swiftmailer configuration.
+$config['swiftmailer.transport']['sendmail_mode'] = 't';
```
Comment (steveworley) [inline drupal/settings/all.settings.php]:
```suggestion
```

**Review by steveworley — APPROVED** [review]

**Comment by murraywoodman** [comment]
There is at least one very large GovCMS clients with skin in the game on this one. In many cases Migrate is not a suitable option for data import, and in these cases running a GET request to pull the data is the only way to get the data in. If this avenue is removed, important functionality will break. This is therefore not a trivial proposal. The security threat should be better explained and clients who are relying on this should be notified. Alternative options for import should be provided.

**Pull request merged**